src/fft_add.h

```diff
@@ -122,55 +127,28 @@ Additive<T>::Additive(const gf::Field<T>& gf, T m, vec::Vector<T>* betas)
     this->beta_m = betas->get(m - 1);
     this->inv_beta_m = this->gf->inv(this->beta_m);
 
-    this->beta_m_powers = new vec::Vector<T>(gf, this->n);
+    this->beta_m_powers = std::make_unique<vec::Vector<T>>(gf, this->n);
     this->compute_beta_m_powers();
 
     this->m_k = arith::exp2<T>(m - 1);
-    this->g0 = new vec::Vector<T>(gf, this->m_k);
-    this->g1 = new vec::Vector<T>(gf, this->m_k);
+    this->g0 = std::make_unique<vec::Vector<T>>(gf, this->m_k);
+    this->g1 = std::make_unique<vec::Vector<T>>(gf, this->m_k);
 
     if (m > 1) {
         // compute gammas and deltas
-        this->gammas = new vec::Vector<T>(gf, m - 1);
-        this->deltas = new vec::Vector<T>(gf, m - 1);
-        this->G = new vec::Vector<T>(gf, this->m_k);
+        this->gammas = std::make_unique<vec::Vector<T>>(gf, m - 1);
+        this->deltas = std::make_shared<vec::Vector<T>>(gf, m - 1);
+        this->G = std::make_unique<vec::Vector<T>>(gf, this->m_k);
         this->compute_basis();
 
-        this->u = new vec::Vector<T>(gf, this->m_k);
-        this->v = new vec::Vector<T>(gf, this->m_k);
+        this->u = std::make_unique<vec::Vector<T>>(gf, this->m_k);
+        this->v = std::make_unique<vec::Vector<T>>(gf, this->m_k);
 
-        this->mem = new vec::Vector<T>(gf, this->n);
-        this->fft_add = new Additive(gf, m - 1, this->deltas);
+        this->mem = std::make_unique<vec::Vector<T>>(gf, this->n);
+        this->fft_add = std::make_unique<Additive>(gf, m - 1, this->deltas);
```

Comment:
Is that why you need `shared_ptr` for `deltas`? I think you could pass a reference here because since the `fft_add` is a member of this class, it is guaranteed to have a shorter lifetime than `this`, so we are sure that it won't outlive `this` (and thus `deltas`), no?

Reply:
`deltas` will be used as `betas` in `this->fft_add`. And we want to use a smart pointer for `betas`. Is there a better solution for `betas` and `deltas`?

Reply:
OK for `shared_pointer` since it's a kind of recursive classes and I don't have any better idea right now.

Reply:
yes, it is. Thanks 👍